tests/build-std/main.rs

```diff
@@ -227,3 +227,50 @@ fn custom_test_framework() {
         .build_std_arg("core")
         .run();
 }
+
+/// like cross-custom but uses per-package-target instead
+#[cargo_test(build_std_real)]
+fn per_package_target() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                cargo-features = ["per-package-target"]
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2018"
+                default-target = "custom-target.json"
```

Comment:
I don't think this is testing this target. Because build-std requires --target to be specified, the default target is never used.  I might suggest using `forced-target`.